dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -749,20 +753,90 @@ def test_json_encode_after_update(self):
         serialized = json.dumps(profile, cls=ProfileEncoder)
         expected = json.dumps(
             {
-                "name": None,
-                "col_index": np.nan,
-                "sample_size": 12,
-                "metadata": {},
-                "times": {"categories": 0.0},
-                "thread_safe": True,
-                "_categories": {"c": 5, "b": 4, "a": 3},
-                "_CategoricalColumn__calculations": {},
-                "_top_k_categories": None,
-            },
+                "class": "CategoricalColumn",
+                "data": {
+                    "name": None,
+                    "col_index": np.nan,
+                    "sample_size": 12,
+                    "metadata": {},
+                    "times": {"categories": 0.0},
+                    "thread_safe": True,
+                    "_categories": {"c": 5, "b": 4, "a": 3},
+                    "_CategoricalColumn__calculations": {},
+                    "_top_k_categories": None,
+                },
+            }
         )
 
         self.assertEqual(serialized, expected)
 
+    def test_json_decode(self):
+        serialized_json = json.dumps(
+            {
+                "class": "CategoricalColumn",
+                "data": {
+                    "name": "0",
+                    "col_index": np.nan,
+                    "sample_size": 0,
+                    "metadata": dict(),
+                    "times": defaultdict(),
+                    "thread_safe": True,
+                    "_categories": defaultdict(int),
+                    "_CategoricalColumn__calculations": dict(),
+                    "_top_k_categories": None,
+                },
+            }
+        )
+
+        deserialized = decode_column_profiler(serialized_json)
+        exepcted = CategoricalColumn("0")
+
+        test_utils.assert_profiles_equal(deserialized, exepcted)
+
+    def test_json_decode_after_update(self):
+        # Actual deserialization
+        serialized = json.dumps(
```

Comment:
by not calc'ing serialized from `df_categorical`, i'm concerned they could become disconnected.

it might be better todo this more similar to the other saves like:
```python
serialized  = json.dumps(df_categorical)
```

Reply:
then we shouldn't have to reset naming as well.

Reply:
Do you mean something like

```py
expected_profile = CategoricalColumn(df_categorical.name)
expected_profile.update(df_categorical)

serialized = json.dumps(expected_profile)
deserialized = decode_column_profile(serialized)

test_utils.assert_profiles_equal(deserialized, expected_profile)
```

Reply:
yes, I think this is a better methodology

Reply:
yes